src/include/OSL/mask.h

```diff
@@ -156,35 +156,38 @@ template<int WidthT> class Mask {
     // NOTE: undefined result if no bits are on
     OSL_FORCEINLINE int first_on() const
     {
-        // FIXME(C++20)
-        // return std::countr_zero(m_value);
-
-        if (value_width <= 32) {
-#if __INTEL_COMPILER
+#if OSL_CPLUSPLUS_VERSION >= 20
+        return std::countr_zero(m_value);
+#elif OSL_INTEL_COMPILER
+        if (value_width <= 32)
             return _bit_scan_forward(m_value);
-#else
+        else
+            return _bit_scan_forward(m_value);  // Correct?
```

Comment:
probably not correct, that intrinsic only takes 32 bit int.  Looks like:
unsigned char _BitScanForward64 (unsigned __int32* index, unsigned __int64 a) is needed, 
```
            unsigned __int32 index;
            _BitScanForward64(&index, m_value);
            return static_cast<int>(index);
```
should do it.

Reply:
Thanks!

Reply:
Wait, I thought _BitScanForward[64] for MSVS? Does that work on Intel, too?

Reply:
https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=scan&expand=392,393,406,409

I think so

Reply:
Well, that even makes it look like I can collapse the INTEL and MSVC clauses to just use the same calls.

Reply:
Well, that would mean calling _BitScanForward(&index, m_value); instead of _bit_scan_forward
and in the end maybe not practical difference.  But as defined the intrinsics are different, _BitScanForward checks the incoming value for 0 and returns that, whereas _bit_scan_forward is undefined for 0, but because we don't do anything with the return value of _BitScanForward, the compiler optimizes the extra check away.

So it would work either way.  But this call "first_on" is undefined for 0, that is why I preferred the _bit_scan_forward.  In the end it probably doesn't make a difference one way or another.